File: netlogo-gui/src/main/prim/_setletvariable.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.core.Let;
import org.nlogo.api.LogoException;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;

public final strictfp class _setletvariable
    extends Command {
  public final Let let;
  private final String name;

  public _setletvariable(_letvariable original) {
    let = original.let;
    name = original.name;
  }

  @Override
  public void perform(final Context context) throws LogoException {
    context.setLet(let, args[0].report(context));
    context.ip = next;
  }



  @Override
  public String toString() {
    return super.toString() + ":" + name;
  }
}

File: netlogo-gui/src/main/prim/_setletvariable.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.core.Let
import org.nlogo.api.LogoException
import org.nlogo.nvm.{ Command, Context }

class _setletvariable(private[this] val _let: Let) extends Command {
  def this(original: _letvariable) = this(original.let)

  def let = _let

  @throws(classOf[LogoException])
  override def perform(context: Context): Unit = {
    perform_1(context, args(0).report(context))
  }

  def perform_1(context: Context, value: AnyRef): Unit = {
    context.activation.binding.setLet(_let, value)
    context.ip = next
  }

  override def toString: String =
    s"${super.toString}:${let.name}"
}
